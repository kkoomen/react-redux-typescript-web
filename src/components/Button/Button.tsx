import React from 'react';

interface Props {
  children?: any,
};
interface State {};

class Button extends React.Component<Props, State> {
  state: State = {};

  render(): React.ReactNode {
    const { children, ...props } = this.props;
    return (
      <button type="button" className="Button">{children}</button>
    );
  }
}

export default Button
