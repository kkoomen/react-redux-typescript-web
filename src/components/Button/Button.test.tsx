import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Button from './Button';

describe('Button', () => {
  it('should match its snapshot', () => {
    const element = shallow(<Button />);
    expect(toJson(element)).toMatchSnapshot();
  });

  it('should render with label', () => {
    const label = 'Lorem ipsum';
    const element = shallow(<Button>{label}</Button>);
    expect(element.text()).toBe(label);
  })
});
