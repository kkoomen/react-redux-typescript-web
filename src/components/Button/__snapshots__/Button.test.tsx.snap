// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match its snapshot 1`] = `
<button
  className="Button"
  type="button"
/>
`;
